packages/react-core/src/components/BackToTop/__tests__/BackToTop.test.tsx

```diff
@@ -1,22 +1,160 @@
+});
+
+test('Renders with custom className prop passed', () => {
+  render(<BackToTop className="test-class" />);
+  expect(screen.getByRole(`button`).parentElement).toHaveClass('test-class');
+});
+
+test('Renders with custom title prop passed', () => {
+  render(<BackToTop title="Back to the Future" />);
+  expect(screen.getByRole(`button`).parentElement).toHaveTextContent('Back to the Future');
+});
+
+test('Renders with passed aria-label', () => {
+  render(<BackToTop aria-label="Back to top test" />);
+  expect(screen.getByRole(`button`).parentElement).toHaveAccessibleName('Back to top test');
+});
+
+test('BackToTop can be accessed via passed innerRef', () => {
+  const testRef: RefObject<HTMLElement> = React.createRef();
+  render(<BackToTop innerRef={testRef} isAlwaysVisible />);
+  global.scrollTo = jest.fn();
+  testRef.current?.click();
+  expect(global.scrollTo).toBeCalledTimes(1);
+});
+
+test('BackToTop reacts to scrolling inside element passed via scrollableSelector', () => {
+  render(
+    <div id="backToTopWrapper">
+      <BackToTop scrollableSelector="#backToTopWrapper" />
+    </div>
+  );
+  const wrapper = document.getElementById('backToTopWrapper');
+  fireEvent.scroll(wrapper as HTMLElement, { target: { scrollY: 401 } });
+
+  expect(screen.getByRole(`button`).parentElement).not.toHaveClass('pf-m-hidden');
+});
+
+test('BackToTop does not react to scrolling inside window when scrollableSelector passed', () => {
+  render(
+    <div id="backToTopWrapper">
+      <BackToTop scrollableSelector="#backToTopWrapper" />
+    </div>
+  );
+  fireEvent.scroll(window, { target: { scrollY: 401 } });
+
+  expect(screen.getByRole(`button`).parentElement).toHaveClass('pf-m-hidden');
+});
+
+test('Clicking backToTop scrolls back to top of the element passed via scrollableSelector', async () => {
+  render(
+    <div id="backToTopWrapper">
+      <BackToTop scrollableSelector="#backToTopWrapper" />
+    </div>
+  );
+  const user = userEvent.setup();
+  const wrapper = document.getElementById('backToTopWrapper');
+  fireEvent.scroll(wrapper as HTMLElement, { target: { scrollY: 401 } });
+  wrapper!.scrollTo = jest.fn();
+  await user.click(screen.getByRole(`button`).parentElement as Element);
+
+  expect(wrapper?.scrollTo).toBeCalledTimes(1);
+});
+
+test('Passes correct text content to button child component', () => {
+  render(<BackToTop title="Back to the future" />);
+
+  expect(screen.getByText('Back to the future')).toBeVisible();
+});
+
+test('Passes correct variant to button child component', () => {
+  render(<BackToTop title="Back to the future" />);
+
+  expect(screen.getByText('primary')).toBeVisible();
+});
+
+test('Passes correct iconPosition to button child component', () => {
+  render(<BackToTop />);
+
+  expect(screen.getByText('right')).toBeVisible();
+});
+
+test('Passes correct icon to button child component', () => {
+  render(<BackToTop />);
+
+  expect(screen.getByTestId('icon').firstChild).toBeTruthy();
+});
```

Comment:
Nit: for the icon I would rather see us do something like what I did [for the angle right icon in the accordion toggle](https://github.com/patternfly/patternfly-react/blob/910d0ce39921d2f89298f0fca3694ea6dc141b63/packages/react-core/src/components/Accordion/__tests__/AccordionToggle.test.tsx#L8) where we mock it out (since the SVG isn't actually a responsibility of BackToTop), and then I would just assert that the icon div contains the text you pass in for the mock so that we know it's being sent to Button properly.